### PR TITLE
[#8480] add global navigation side bar

### PR DESCRIPTION
**Pull request by BillionaireDY**
@binDongKim

### CODE REVIEW

File: web/src/main/angular/src/app/core/components/side-navigation-bar/side-navigation-bar-container.component.ts [hunk 1]
```diff
@@ -0,0 +1,183 @@
+import { Component, OnInit, ChangeDetectionStrategy, ChangeDetectorRef, OnDestroy } from '@angular/core';
+import { Router } from '@angular/router';
+import { UrlPathId } from 'app/shared/models';
+import { WebAppSettingDataService, NewUrlStateNotificationService, StoreHelperService, TRACKED_EVENT_LIST, UrlRouteManagerService, WindowRefService, AnalyticsService } from 'app/shared/services';
+import { Actions } from 'app/shared/store/reducers';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
```
Comment (binDongKim) [inline web/src/main/angular/src/app/core/components/side-navigation-bar/side-navigation-bar-container.component.ts]:
```suggestion
import { Component, OnInit, ChangeDetectionStrategy, ChangeDetectorRef, OnDestroy } from '@angular/core';
import { Router } from '@angular/router';
import { Subject } from 'rxjs';
import { takeUntil } from 'rxjs/operators';

import { UrlPathId } from 'app/shared/models';
import { WebAppSettingDataService, NewUrlStateNotificationService, StoreHelperService, TRACKED_EVENT_LIST, UrlRouteManagerService, WindowRefService, AnalyticsService } from 'app/shared/services';
import { Actions } from 'app/shared/store/reducers';
```

Locate the external import statement at the top and the others after them(with a blank line)

File: web/src/main/angular/src/app/core/components/side-navigation-bar/side-navigation-bar-container.component.ts [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    class?: string;
+    iconClass?: string;
+    disable?: () => boolean;
+    onClick?: (item: ISNBItem) => void;
+    childItems?: ISNBItem[];
+}
+
+interface ISNBMeta {
+    topItems:ISNBItem[], bottomItems: ISNBItem[],
+}
```
Comment (binDongKim) [inline web/src/main/angular/src/app/core/components/side-navigation-bar/side-navigation-bar-container.component.ts]:
```suggestion
interface ISNBMeta {
    topItems: ISNBItem[];
    bottomItems: ISNBItem[];
}
```

Correct the interface syntax

File: web/src/main/angular/src/app/core/components/configuration-experimental/configuration-experimental-container.component.ts [hunk 3]
```diff
@@ -67,6 +69,12 @@ export class ConfigurationExperimentalContainerComponent implements OnInit {
                 this.webAppSettingDataService.setExperimentalOption('scatterSampling', this.sampleScatter);
                 this.applyGA({optionKey, optionValue: this.sampleScatter});
                 break;
+            case 'sideNavigationUI':
+                this.sideNavigationUI = !this.sideNavigationUI;
+                this.webAppSettingDataService.setExperimentalOption('sideNavigationUI', this.sideNavigationUI);
+                this.applyGA({optionKey, optionValue: this.sideNavigationUI});
+                window.location.reload();
```
Comment (binDongKim) [inline web/src/main/angular/src/app/core/components/configuration-experimental/configuration-experimental-container.component.ts]:
```suggestion
this.urlRouteManagerService.reload();
```
Use `UrlRouteManagerService` reload method: `this.urlRouteManagerService.reload();`

File: web/src/main/angular/src/app/core/components/side-navigation-bar/side-navigation-bar-container.component.ts [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SideNavigationBarContainerComponent implements OnInit, OnDestroy {
+    private unsubscribe = new Subject<void>();
+    private selectedAgent = '';
+  
+    basePath = '';
+    enableRealTime: boolean;
+    logoPath: string;
+    minimize: boolean = false;
```
Comment (binDongKim) [inline web/src/main/angular/src/app/core/components/side-navigation-bar/side-navigation-bar-container.component.ts]:
```suggestion
    minimize = false;
```
Delete the type annotation per lint

File: web/src/main/angular/src/app/core/components/side-navigation-bar/side-navigation-bar-container.component.ts [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+        private cd: ChangeDetectorRef,
+        private windowRefService: WindowRefService,
+        private analyticsService: AnalyticsService,
+        private storeHelperService: StoreHelperService,
+        private urlRouteManagerService: UrlRouteManagerService,
+        private webAppSettingDataService: WebAppSettingDataService,
+        private newUrlStateNotificationService: NewUrlStateNotificationService,
+    ) {
+        this.basePath = this.router.url;
+        this.minimize = this.webAppSettingDataService.getSideNavBarScale();
```
Comment (binDongKim) [inline web/src/main/angular/src/app/core/components/side-navigation-bar/side-navigation-bar-container.component.ts]: Initialize variables in `ngOnInit()`. There shouldn't be any initialization in constructor unless it's necessary.

File: web/src/main/angular/src/app/core/components/side-navigation-bar/side-navigation-item.component.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
+import { Component, OnInit, Input, Output, EventEmitter, SimpleChanges, ViewChild, ViewChildren, QueryList, ElementRef, OnChanges, AfterViewInit } from '@angular/core';
```
Comment (binDongKim) [inline web/src/main/angular/src/app/core/components/side-navigation-bar/side-navigation-item.component.ts]: Add a line-break after the external import statement

File: web/src/main/angular/src/app/core/components/side-navigation-bar/side-navigation-item.component.ts [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+    constructor(
+        private analyticsService: AnalyticsService,
+    ) {}
+    
+    ngOnInit() {
+    }
+
+    ngOnChanges() {
+    }
```
Comment (binDongKim) [inline web/src/main/angular/src/app/core/components/side-navigation-bar/side-navigation-item.component.ts]: Remove unused lifecycle hook

File: web/src/main/angular/src/app/routes/filtered-map-page/filtered-map-page.component.ts [hunk 8]
```diff
@@ -1,4 +1,5 @@
 import { Component, OnInit, ChangeDetectionStrategy } from '@angular/core';
```
Comment (binDongKim) [inline web/src/main/angular/src/app/routes/filtered-map-page/filtered-map-page.component.ts:1]: Add a line-break after the external import statement

**Review by binDongKim — CHANGES_REQUESTED** [review]

File: web/src/main/angular/src/app/core/components/side-navigation-bar/side-navigation-bar-container.component.ts [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+    iconClass?: string;
+    disable?: () => boolean;
+    onClick?: (item: ISNBItem) => void;
+    childItems?: ISNBItem[];
+}
+
+interface ISNBMeta {
+    topItems:ISNBItem[], 
+    bottomItems: ISNBItem[],
+}
```
Comment (binDongKim) [inline web/src/main/angular/src/app/core/components/side-navigation-bar/side-navigation-bar-container.component.ts]:
```suggestion
interface ISNBMeta {
    topItems: ISNBItem[];
    bottomItems: ISNBItem[];
}
```

Correct the interface syntax

File: web/src/main/angular/src/app/core/components/side-navigation-bar/side-navigation-item.component.ts [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+    ngOnInit() {
+    }
+
+    ngAfterViewInit() {
+        this.positionLinkLayer()
+    }
+
+    ngAfterViewChecked() {
+        this.positionLinkLayer()
+    }
```
Comment (binDongKim) [inline web/src/main/angular/src/app/core/components/side-navigation-bar/side-navigation-item.component.ts:41]: Remove if it's unnecessary

**Review by binDongKim — CHANGES_REQUESTED** [review]

File: web/src/main/angular/src/app/core/components/side-navigation-bar/side-navigation-item.component.ts (hunk 10 above)
Comment (BillionaireDY) [reply web/src/main/angular/src/app/core/components/side-navigation-bar/side-navigation-item.component.ts:41]: This is necessary to calculate the height when SNB scale changes.

**Pull request merged**